Bottom: 388ca37ef3215836da13e5a3b4259703c98c377b
Top:    c43ee60463f8965f259bff46add1c303340fefed
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 14:57:22 +0000

Refresh of remove-typeptr-and-typehandle-

---

diff --git a/src/buffer.c b/src/buffer.c
index 03c2cfd..0394f2f 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -84,7 +84,7 @@ static void SwapBuffer (BufferType *);
 /* ---------------------------------------------------------------------------
  * some local identifiers
  */
-static DataType *Dest, Source;
+static DataType *Dest, *Source;
 
 static ObjectFunctionType AddBufferFunctions = {
   AddLineToBuffer,
@@ -861,7 +861,7 @@ SmashBufferElement (BufferType *Buffer)
 {
   ElementType *element;
   Cardinal group;
-  LayerType *clayer, slayer;
+  LayerType *clayer, *slayer;
 
   if (Buffer->Data->ElementN != 1)
     {
diff --git a/src/copy.c b/src/copy.c
index 4fda96c..9d92ced 100644
--- a/src/copy.c
+++ b/src/copy.c
@@ -333,8 +333,8 @@ CopyPastebufferToLayout (Coord X, Coord Y)
   /* paste all layers */
   for (i = 0; i < max_copper_layer + 2; i++)
     {
-      LayerType *sourcelayer = &PASTEBUFFER->Data->Layer[i],
-	destlayer = LAYER_PTR (i);
+      LayerType *sourcelayer = &PASTEBUFFER->Data->Layer[i];
+      LayerType *destlayer = LAYER_PTR (i);
 
       if (destlayer->On)
 	{
diff --git a/src/crosshair.c b/src/crosshair.c
index 485a8d1..f0526b8 100644
--- a/src/crosshair.c
+++ b/src/crosshair.c
@@ -496,7 +496,7 @@ XORDrawMoveOrCopyObject (void)
   ptr = Crosshair.AttachedObject.Rubberband;
   while (i)
     {
-      PointType *point1, point2;
+      PointType *point1, *point2;
 
       if (TEST_FLAG (VIAFLAG, ptr->Line))
 	{
diff --git a/src/find.c b/src/find.c
index 801ce16..8c737a3 100644
--- a/src/find.c
+++ b/src/find.c
@@ -278,8 +278,7 @@ typedef struct
   Cardinal Location,            /* currently used position */
     DrawLocation, Number,       /* number of objects in list */
     Size;
-}
-ListType, *ListType *;
+} ListType;
 
 /* ---------------------------------------------------------------------------
  * some local identifiers
diff --git a/src/fontmode.c b/src/fontmode.c
index ad30665..e09f825 100644
--- a/src/fontmode.c
+++ b/src/fontmode.c
@@ -76,7 +76,7 @@ FontEdit (int argc, char **argv, Coord Ux, Coord Uy)
 {
   FontType *font;
   SymbolType *symbol;
-  LayerType *lfont, lorig, lwidth, lgrid;
+  LayerType *lfont, *lorig, *lwidth, *lgrid;
   int s, l;
 
   if (hid_actionl ("New", "Font", 0))
@@ -180,7 +180,7 @@ FontSave (int argc, char **argv, Coord Ux, Coord Uy)
   SymbolType *symbol;
   int i;
   GList *ii;
-  LayerType *lfont, lwidth;
+  LayerType *lfont, *lwidth;
 
   font = &PCB->Font;
   lfont = PCB->Data->Layer + 0;
diff --git a/src/intersect.c b/src/intersect.c
index bad6c59..59aa064 100644
--- a/src/intersect.c
+++ b/src/intersect.c
@@ -214,7 +214,7 @@ ComputeIntersectionArea (BoxListType *boxlist)
 double
 ComputeUnionArea (BoxListType *boxlist)
 {
-  BoxType **rectLeft, *rectRight;
+  BoxType **rectLeft, **rectRight;
   Cardinal i, j;
   LocationList yCoords;
   SegmentTree segtree;
@@ -283,13 +283,15 @@ ComputeUnionArea (BoxListType *boxlist)
 static int
 compareleft (const void *ptr1, const void *ptr2)
 {
-  BoxType **b1 = (BoxType **) ptr1, *b2 = (BoxType **) ptr2;
+  BoxType **b1 = (BoxType **) ptr1;
+  BoxType **b2 = (BoxType **) ptr2;
   return (*b1)->X1 - (*b2)->X1;
 }
 static int
 compareright (const void *ptr1, const void *ptr2)
 {
-  BoxType **b1 = (BoxType **) ptr1, *b2 = (BoxType **) ptr2;
+  BoxType **b1 = (BoxType **) ptr1;
+  BoxType **b2 = (BoxType **) ptr2;
   return (*b1)->X2 - (*b2)->X2;
 }
 static int
diff --git a/src/misc.c b/src/misc.c
index 5a4ef9c..f675079 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -342,7 +342,7 @@ void
 SetElementBoundingBox (DataType *Data, ElementType *Element,
                        FontType *Font)
 {
-  BoxType *box, vbox;
+  BoxType *box, *vbox;
 
   if (Data && Data->element_tree)
     r_delete_entry (Data->element_tree, (BoxType *) Element);
diff --git a/src/polygon.c b/src/polygon.c
index f865da2..3311b09 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1271,7 +1271,7 @@ Cardinal
 GetLowestDistancePolygonPoint (PolygonType *Polygon, Coord X, Coord Y)
 {
   double mindistance = (double) MAX_COORD * MAX_COORD;
-  PointType *ptr1, ptr2;
+  PointType *ptr1, *ptr2;
   Cardinal n, result = 0;
 
   /* we calculate the distance to each segment and choose the
diff --git a/src/rats.c b/src/rats.c
index a669f36..e7221fa 100644
--- a/src/rats.c
+++ b/src/rats.c
@@ -450,7 +450,7 @@ CheckShorts (LibraryMenuType *theNet)
 static bool
 GatherSubnets (NetListType *Netl, bool NoWarn, bool AndRats)
 {
-  NetType *a, b;
+  NetType *a, *b;
   ConnectionType *conn;
   Cardinal m, n;
   bool Warned = false;
@@ -558,12 +558,12 @@ DrawShortestRats (NetListType *Netl, void (*funcp) (register ConnectionType *, r
 {
   RatType *line;
   register float distance, temp;
-  register ConnectionType *conn1, conn2, firstpoint, secondpoint;
+  register ConnectionType *conn1, *conn2, *firstpoint, *secondpoint;
   PolygonType *polygon;
   bool changed = false;
   bool havepoints;
   Cardinal n, m, j;
-  NetType *next, subnet, theSubnet = NULL;
+  NetType *next, *subnet, *theSubnet = NULL;
 
   /* This is just a sanity check, to make sure we're passed
    * *something*.
@@ -725,7 +725,7 @@ DrawShortestRats (NetListType *Netl, void (*funcp) (register ConnectionType *, r
 bool
 AddAllRats (bool SelectedOnly, void (*funcp) (register ConnectionType *, register ConnectionType *, register RouteStyleType *))
 {
-  NetListType *Nets, Wantlist;
+  NetListType *Nets, *Wantlist;
   NetType *lonesome;
   ConnectionType *onepin;
   bool changed, Warned = false;
@@ -820,7 +820,7 @@ NetListListType
 CollectSubnets (bool SelectedOnly)
 {
   NetListListType result = { 0, 0, NULL };
-  NetListType *Nets, Wantlist;
+  NetListType *Nets, *Wantlist;
   NetType *lonesome;
   ConnectionType *onepin;
 
diff --git a/src/search.c b/src/search.c
index c66fa28..5f4ea5e 100644
--- a/src/search.c
+++ b/src/search.c
@@ -325,7 +325,7 @@ SearchRatLineByLocation (int locked, RatType ** Line, RatType ** Dummy1,
  */
 struct arc_info
 {
-  ArcType **Arc, *Dummy;
+  ArcType **Arc, **Dummy;
   PointType **Point;
   double least;
   jmp_buf env;
diff --git a/src/undo.c b/src/undo.c
index 1e7df42..ca41c74 100644
--- a/src/undo.c
+++ b/src/undo.c
@@ -81,14 +81,12 @@ static bool added_undo_between_increment_and_restore = false;
 typedef struct			/* information about a change command */
 {
   char *Name;
-}
-ChangeNameType, *ChangeNameType *;
+} ChangeNameType;
 
 typedef struct			/* information about a move command */
 {
   Coord DX, DY;		/* movement vector */
-}
-MoveType, *MoveType *;
+} MoveType;
 
 typedef struct			/* information about removed polygon points */
 {
@@ -96,42 +94,36 @@ typedef struct			/* information about removed polygon points */
   int ID;
   Cardinal Index;		/* index in a polygons array of points */
   bool last_in_contour;		/* Whether the point was the last in its contour */
-}
-RemovedPointType, *RemovedPointType *;
+} RemovedPointType;
 
 typedef struct			/* information about rotation */
 {
   Coord CenterX, CenterY;	/* center of rotation */
   Cardinal Steps;		/* number of steps */
-}
-RotateType, *RotateType *;
+} RotateType;
 
 typedef struct			/* information about moves between layers */
 {
   Cardinal OriginalLayer;	/* the index of the original layer */
-}
-MoveToLayerType, *MoveToLayerType *;
+} MoveToLayerType;
 
 typedef struct			/* information about layer changes */
 {
   int old_index;
   int new_index;
-}
-LayerChangeType, *LayerChangeType *;
+} LayerChangeType;
 
 typedef struct			/* information about poly clear/restore */
 {
   bool Clear;		/* true was clear, false was restore */
   LayerType *Layer;
-}
-ClearPolyType, *ClearPolyType *;
+} ClearPolyType;
 
 typedef struct			/* information about netlist lib changes */
 {
   LibraryType *old;
   LibraryType *lib;
-}
-NetlistChangeType, *NetlistChangeType *;
+} NetlistChangeType;
 
 typedef struct			/* holds information about an operation */
 {
@@ -154,8 +146,7 @@ typedef struct			/* holds information about an operation */
     long int CopyID;
   }
   Data;
-}
-UndoListType, *UndoListType *;
+} UndoListType;
 
 /* ---------------------------------------------------------------------------
  * some local variables
@@ -873,7 +864,7 @@ UndoNetlistChange (UndoListType *Entry)
 {
   NetlistChangeType *l = & Entry->Data.NetlistChange;
   unsigned int i, j;
-  LibraryType *lib, saved;
+  LibraryType *lib, *saved;
 
   lib = l->lib;
   saved = l->old;
